### PR TITLE
Python: add IDs with slot index method

### LINKED ISSUE #1947: bindings: add IDs slot index getter

### PR DESCRIPTION
**Pull request by Thoralf-M**
# Description of change

Add IDs with slot index method

## Links to any relevant issues

https://github.com/iotaledger/iota-sdk/issues/1947

## Type of change

- Breaking change (fix or feature that would cause existing functionality to not work as expected)

## How the change has been tested

Describe the tests that you ran to verify your changes.

Make sure to provide instructions for the maintainer as well as any relevant configurations.

### CODE REVIEW

File: bindings/python/iota_sdk/types/output_id.py [hunk 1]
```diff
@@ -14,19 +15,19 @@ class OutputId(dict):
...
+                'transaction_id length must be 74 characters with 0x prefix')
         if not transaction_id.startswith('0x'):
             raise ValueError('transaction_id must start with 0x')
         # Validate that it has only valid hex characters
         int(transaction_id[2:], 16)
-        output_index_hex = (output_index).to_bytes(4, "little").hex()
+        output_index_hex = (output_index).to_bytes(2, "little").hex()
         self.output_id = transaction_id + output_index_hex
-        self.transaction_id = transaction_id
+        self.transaction_id = TransactionId(transaction_id)
```
Comment (DaughterOfMars) [inline bindings/python/iota_sdk/types/output_id.py]: is this double wrapping?

Comment (Thoralf-M) [inline bindings/python/iota_sdk/types/output_id.py]:
```suggestion
        self.transaction_id = transaction_id
```

Comment (Thoralf-M) [reply bindings/python/iota_sdk/types/output_id.py]: Yes good catch

File: bindings/python/iota_sdk/types/ids.py
Comment (Alex6323) [inline bindings/python/iota_sdk/types/ids.py:1]: Are we sure we want to put all IDs into one file and not have them with their corresponding types? `BlockId` ~> `block.py` etc? If we really want to have something as a common `ids.py` then I'ld prefer a re-export. At least that's how you could do that in Rust.

**Review by thibault-martinez — COMMENTED** [review]
HexStr can be replaced in more places:
- [x] `iota-sdk/bindings/python/iota_sdk/client/_high_level_api.py`
- [x] `iota-sdk/bindings/python/iota_sdk/client/_node_core_api.py`
- [x] `iota-sdk/bindings/python/iota_sdk/client/_utils.py`
- [x] `iota-sdk/bindings/python/iota_sdk/types/balance.py`
- [x] `iota-sdk/bindings/python/iota_sdk/types/block_builder_options.py`
- [x] `iota-sdk/bindings/python/iota_sdk/types/node_info.py`
- [x] `iota-sdk/bindings/python/iota_sdk/types/output_data.py`
- [x] `iota-sdk/bindings/python/iota_sdk/types/block/block.py`
- [x] `iota-sdk/bindings/python/iota_sdk/types/block/body/basic.py`
- [x] `iota-sdk/bindings/python/iota_sdk/types/block/body/validation.py`
- [x]  `iota-sdk/bindings/python/iota_sdk/wallet/wallet.py`

File: bindings/python/iota_sdk/types/ids.py
Comment (Thoralf-M) [reply bindings/python/iota_sdk/types/ids.py:1]: Moved them

**Comment by thibault-martinez** [comment]
> HexStr can be replaced in more places:
> 
> * [x]  `iota-sdk/bindings/python/iota_sdk/client/_high_level_api.py`
> * [x]  `iota-sdk/bindings/python/iota_sdk/client/_node_core_api.py`
> * [x]  `iota-sdk/bindings/python/iota_sdk/client/_utils.py`
> * [x]  `iota-sdk/bindings/python/iota_sdk/types/balance.py`
> * [x]  `iota-sdk/bindings/python/iota_sdk/types/block_builder_options.py`
> * [x]  `iota-sdk/bindings/python/iota_sdk/types/node_info.py`
> * [x]  `iota-sdk/bindings/python/iota_sdk/types/output_data.py`
> * [x]  `iota-sdk/bindings/python/iota_sdk/types/block/block.py`
> * [x]  `iota-sdk/bindings/python/iota_sdk/types/block/body/basic.py`
> * [x]  `iota-sdk/bindings/python/iota_sdk/types/block/body/validation.py`
> * [x]   `iota-sdk/bindings/python/iota_sdk/wallet/wallet.py`

Missing `get_tips` (even though we'll remove...) and `announce_candidacy`

File: bindings/python/iota_sdk/types/context_input.py [hunk 2]
```diff
@@ -5,7 +5,12 @@
 from dataclasses import dataclass, field
 from enum import IntEnum
 from typing import Any, Dict, List, TypeAlias, Union
-from iota_sdk.types.common import HexStr, json
+from iota_sdk.types.common import HexStr, IdWithSlotIndex, json
+
+
+class SlotCommitmentId(IdWithSlotIndex):
```
Comment (thibault-martinez) [inline bindings/python/iota_sdk/types/context_input.py]: Not too sure about this being here...

Comment (Thoralf-M) [reply bindings/python/iota_sdk/types/context_input.py]: Me neither, but don't know where else it should be

Comment (Thoralf-M) [reply bindings/python/iota_sdk/types/context_input.py]: Should I put it in common.py?

Comment (thibault-martinez) [reply bindings/python/iota_sdk/types/context_input.py]: Alex added `bindings/python/iota_sdk/types/slot.py` for the commitment itself, maybe also create this file and he'll add it to it

**Review by thibault-martinez — CHANGES_REQUESTED** [review]
Failing CI

**Comment by Alex6323** [comment]
> Failing CI

Someone did not run `tox -e all` before pushing :trollface:

**Comment by Thoralf-M** [comment]
@Alex6323 doesn't help :trollface: 
```
ERROR: could not install deps [-rrequirements-dev.txt, pylint, autopep8]; v = InvocationError('could not find executable python', None)
____________________________________________________________________________________________ summary ____________________________________________________________________________________________
ERROR:   all: could not install deps [-rrequirements-dev.txt, pylint, autopep8]; v = InvocationError('could not find executable python', None)
```
But I was actually running `pylint --rcfile=.pylintrc iota_sdk/**/*.py`, but that also didn't show the problems :(

**Review by thibault-martinez — APPROVED** [review]

**Review by Alex6323 — DISMISSED** [review]

**Review by thibault-martinez — CHANGES_REQUESTED** [review]
Failing CI

**Review by thibault-martinez — APPROVED** [review]

**Review by DaughterOfMars — APPROVED** [review]

**Pull request merged**